### PR TITLE
Add support for multiple test command status values

### PR DESCRIPTION
**Pull request by tldahlgren**
This PR allows status values to be lists since not all versions will necessarily return the same command exit code.  It also fixes the error from an assert.

This PR has been tested against a (local) version of the `openmpi` test (see #15456 ).

### CODE REVIEW

File: lib/spack/spack/package.py [hunk 1]
```diff
@@ -1926,23 +1926,31 @@ def run_test(self, exe, options, expected, status):
...
             expected (list of str): list of expected output strings
-            status (int or None): the expected process status if int or None
-                if the test is expected to succeed
+            status (int or list of int): possible passing status values with
+                0 and None meaning the test should succeed
         """
-        result = 'fail with status {0}'.format(status) if status else 'succeed'
-        tty.debug('test: {0}: expect to {1}'.format(exe, result))
+        case = '' if isinstance(status, int) else 'in '
+        tty.msg('test: {0}: expect command status {1}{2}'
```
Comment (becker33) [inline lib/spack/spack/package.py]: Can you leave this as a debug message? I think it's too verbose to be non-conditional.

File: lib/spack/spack/package.py [hunk 2]
```diff
@@ -1926,23 +1926,31 @@ def run_test(self, exe, options, expected, status):
...
+                .format(exe, case, status))
         runner = which(exe)
         assert runner is not None
 
         try:
             output = runner(*options, output=str.split, error=str.split)
-            assert not status, 'Expected execution to fail'
+            assert not status or None in status or 0 in status, \
+                'Expected execution to succeed'
```
Comment (becker33) [inline lib/spack/spack/package.py]: This message should not be changing.

File: lib/spack/spack/package.py [hunk 3]
```diff
@@ -1926,23 +1926,31 @@ def run_test(self, exe, options, expected, status):
             exe (str): the name of the executable
             options (list of str): list of options to pass to the runner
             expected (list of str): list of expected output strings
-            status (int or None): the expected process status if int or None
-                if the test is expected to succeed
+            status (int or list of int): possible passing status values with
+                0 and None meaning the test should succeed
         """
-        result = 'fail with status {0}'.format(status) if status else 'succeed'
```
Comment (becker33) [inline lib/spack/spack/package.py]: I think a lot of this code could be cleaned up by putting at the top of the method:

```
if not isinstance(status, list):
    status = [status]
```
And then we can assume it's a list throughout.

File: lib/spack/spack/package.py (hunk 1 above)
Comment (tldahlgren) [reply lib/spack/spack/package.py:1933]: Oops.  Definitely!

**Comment by tldahlgren** [comment]
@becker33 So I have to use `spack -d test` to get my PASS/FAIL messages in the test output now?  I find the non-debug output a little confusing, at least for the tests I've written.

**Comment by tldahlgren** [comment]
I've tried several things to get rid of the extra commits per our chat exchange but haven't hit the right incantation.

**Comment by tldahlgren** [comment]
The relevant changes were merged into the latest `features/spack-test`.  See #15979.

**Pull request closed without merging**